app/app/vhosts/index/route.js

```diff
@@ -9,5 +9,25 @@ export default Ember.Route.extend({
     if(model.get('length') === 0) {
       return this.transitionTo('app.vhosts.new', this.modelFor('app'));
     }
+  },
+
+  actions: {
+    delete: function(vhost){
+      let op = this.store.createRecord('operation', {
+        type: 'deprovision',
+        vhost: vhost
+      });
+
+      this.controller.set('error', null);
+      vhost.set('isDeleting', true);
```

Comment:
It definitely makes some sense to make a 'delete-vhost' component that can encapsulate some or all of this logic. At minimum it can encapsulate the button disabling and button text changing during save, removing the necessity of this transient `isDeleting` prop on the vhost.

However because there didn't seem to be a good place to stash an error message (if there is one) on/around that button I added an alert at the top of this `vhosts#index` page, which is what the controller's `error` prop is used for.

Since there are multiple semi-disconnected parts of the UI that are in play here it feels most natural to me to do it this way, even with the ugliness of the `isDeleted` prop on the vhost. The alternative seems like it would be having to coordinate actions on the route and the component to synchronize the states between the two.

I'll give it a shot to see how it feels and update this PR with a delete-vhost component.